editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/restconfiguration/RestConfigEditor.java

```diff
@@ -548,6 +561,35 @@ public void widgetSelected(SelectionEvent e) {
 		
 		return client;
 	}
+	
+	private void addComponentDependency(String componentName) {
+		IWorkbench wb = PlatformUI.getWorkbench();
+		IProgressService ps = wb.getProgressService();
+		try {
+			ps.busyCursorWhile(new IRunnableWithProgress() {
+				public void run(IProgressMonitor pm) {
+					IProject project = parentEditor.getDesignEditor().getWorkspaceProject();
+					final CamelModel camelModel = CamelCatalogCacheManager.getInstance().getCamelModelForProject(project, pm);
+					final ComponentManager componentManager = new ComponentManager(camelModel);
+					Set<Component> componentSet = componentManager.getAllComponents();
+					if (!componentSet.isEmpty()) {
```

Comment:
I think finding the components by Id is a great use case that might be useful in other contexts. What do you think to create the a getComponentbyId(String id) in ComponentManager?
It will avoid to loop and have a break; statement in this RunnableWithProgress